spyder/utils/misc.py

```diff
@@ -153,9 +153,21 @@ def remove_backslashes(path):
 
 
 def get_error_match(text):
-    """Return error match"""
-    import re
-    return re.match(r'  File "(.*)", line (\d*)', text)
+    """Check if text contains a Python error."""
+    # For IPython 8+ tracebacks. We check for this first because we assume most
+    # users need to click on file names present on tracebacks in the IPython
+    # console.
+    # Fixes spyder-ide/spyder#20407
+    ipython8_match = re.match(r'  File (.*):(\d*)', text)
+    if ipython8_match is not None:
+        return ipython8_match
+
+    # For regular Python tracebacks and IPython 7 or less.
+    match_python = re.match(r'  File "(.*)", line (\d*)', text)
+    if match_python is not None:
+        return match_python
```

Comment:
@ccordoba12 checking locally, seems like the order here needs to be changed so Spyder can properly detect the error text with IPython 7.x (at least on Windows):

```suggestion
    # For regular Python tracebacks and IPython 7 or less.
    match_python = re.match(r'  File "(.*)", line (\d*)', text)
    if match_python is not None:
        return match_python

    # For IPython 8+ tracebacks. We check for this first because we assume most
    # users need to click on file names present on tracebacks in the IPython
    # console.
    # Fixes spyder-ide/spyder#20407
    ipython8_match = re.match(r'  File (.*):(\d*)', text)
    if ipython8_match is not None:
        return ipython8_match
```

I think this happens since the error text for IPython 7.x also can partially match the rule for IPython 8.x. There the path of the files starts with a driver letter plus `:` so then something like `File "c:\users\dalth\test\fooo.py"` matches the `r'  File (.*):(\d*)` expression used for IPython 8.x. Probably it would be good to check if the order change suggested can affect the functionality on Linux? 🤔

Reply:
> checking locally, seems like the order here needs to be changed so Spyder can properly detect the error text with IPython 7.x (at least on Windows)

Thanks for checking that @dalthviz!

> Probably it would be good to check if the order change suggested can affect the functionality on Linux?

Checked that and your suggestion works well with both IPython 7 and 8. I just made a little change to it to remove the part in my comment that explained why the IPython 8 regex was used first (which is not true anymore).